packages/query-core/src/query.ts

```diff
@@ -49,6 +49,7 @@ export interface QueryState<TData = unknown, TError = unknown> {
   isInvalidated: boolean
   status: QueryStatus
   fetchStatus: FetchStatus
+  meta?: any
```

Comment:
I marked meta as undefined cause its usually is undefined.

Reply:
`meta` is not a good name because we already have `meta` inside the `Query` itself:

https://github.com/TanStack/query/blob/04945f38b3f2ff82a597083bf1a36724f973bc7e/packages/query-core/src/query.ts#L152

could we just put this value inside `fetchMeta` ?

Reply:
It's not that easy to put it on fetchMeta, because when fetchingNextPage called there is no good way to access previous  fetchMeta and this method will overwrite fetchMeta. So I need to change before this line
https://github.com/TanStack/query/blob/89da08fe0723249a86b681e855937787b3d14eb9/packages/query-core/src/infiniteQueryBehavior.ts#L16

and need lot of check to check `fetchOptions` and `fetchOptions.meta` is not undefined and add new meta on it.

Reply:
yeah it would need shallow merging of the meta. My biggest concern is with the word `meta`, as we already have three different "meta" properties:

- the `fetchMeta`
- meta on queryOptions
- meta on mutationOptions

if we can't unify this with fetchMeta, we have to please find a better name 😅